src/transformers/core_model_loading.py

```diff
@@ -378,7 +408,8 @@ def set_param_for_module(
                 pass  # TODO for "local" stuff, it will trigger missmatched no?
             param_value = torch.nn.Parameter(param_value, requires_grad=param_value.is_floating_point())
 
-        if ref is not None and ref.shape != param_value.shape:
+        # skip mismatch for hf_quantizer for now
+        if ref is not None and ref.shape != param_value.shape and hf_quantizer is None:
```

Comment:
why is the shape of the BnbLinear not correct? this I also don't think we want long term no?